kayvee.go

```diff
@@ -33,3 +44,52 @@ func FormatLog(source string, level LogLevel, title string, data map[string]inte
 	data["title"] = title
 	return Format(data)
 }
+
+// SentryClient interface allows mocking for testing
+type sentryClient interface {
+	Capture(packet *raven.Packet, captureTags map[string]string) (eventID string, ch chan error)
+}
+
+// Logger provides an wrapper methods to do logging using kayvee.
+type Logger struct {
+	source       string
+	logger       *log.Logger
+	sentryClient sentryClient
+}
+
+// NewLogger returns a new *kayvee.Logger.
+// source is the value assigned for all logs generated by the logger
+// log.Logger is the underlying logger used. If nil, uses log.New(os.Stderr, "", log.Ldate|log.Ltime|log.Lshortfile)
+// sentryClient is used to optionally route errors to sentry.
+func NewLogger(source string, logger *log.Logger, sentryClient *raven.Client) *Logger {
```

Comment:
I prefer being explicit in this manner because it is more in line with dependency injection. I am fine with versioning it if we add additional arguments.